### DISCUSSION TITLE
Best places to contribute

### DISCUSSION THREAD

**Discussion by DataWaveAnalytics** [CamDavidsonPilon/lifelines#1588] [Q&A] [answered]
Hey community, 

I love we have a library for survival analysis in Python, and I want to contribute to it (I know there is a section with guidelines). I would like to know which parts of the project need more attention from external contributors. I'm open to improving documentation and code base, so please any advice on where to start is greatly appreciated. Thanks!

**Comment by CamDavidsonPilon** [comment, marked as answer]
Hi @DataWaveAnalytics! 

Sorry about the late reply. Thanks for the offer! That's very kind of you. Heres a list of good spots to contribute:

 - Add more "examples and recipes" to [here](https://lifelines.readthedocs.io/en/latest/Examples.html).
 - I really don't like the structure nor performance of how we convert or create time-varying datasets. Specifically functions like "utils.to_long_format", "utils.to_episodic_format", and "utils.add_covariate_to_timeline".
 - Better integration with `forumlaic` library, and more examples that use formulas. 
 - Fixing our test suite. There are a few flakey tests: I'd rather fix them to be nearly perfect, or just delete them.

**Reply by DataWaveAnalytics** [reply]
Thanks @CamDavidsonPilon, I will explore the topics on your list and will start with some contributions following your guidelines. Let me know if something else comes to mind. Cheers
